lib/miq_automation_engine/service_models/miq_ae_service_container_deployment.rb

```diff
@@ -0,0 +1,192 @@
+require 'net/scp'
+
+module MiqAeMethodService
+  class MiqAeServiceContainerDeployment < MiqAeServiceModelBase
+    expose :container_deployment_nodes, :association => true
+    expose :deployed_ems, :association => true
+    expose :deployed_on_ems, :association => true
+    expose :automation_task, :association => true
+    expose :roles_addresses
+    expose :container_nodes_by_role
+    expose :ssh_auth
+    expose :rhsm_auth
+
+    def assign_container_deployment_node(vm_id, role)
+      container_nodes_by_role(role).each do |deployment_node|
+        next unless deployment_node.vm_id.nil?
+        deployment_node.add_vm vm_id
+      end
+    end
+
+    def subscribe_deployment_master(deployment_master, user)
+      rhsub_user = rhsm_auth.userid
+      rhsub_pass = rhsm_auth.password
+      rhsm_sku = rhsm_auth.rhsm_sku
+      subscribed = perform_agent_commands(deployment_master, user, ["sudo subscription-manager register --username='#{rhsub_user}'  --password='#{rhsub_pass}'"])[:stdout]
+      unless subscribed.include? "This system is already registered"
+        pool_id = perform_agent_commands(deployment_master, user, ["sudo subscription-manager list --available --matches='#{rhsm_sku}' --pool-only"])[:stdout].split("\n").first.delete("\r")
+        perform_agent_commands(deployment_master, user, ["sudo subscription-manager attach --pool='#{pool_id}'",
+                                                         'sudo subscription-manager repos --disable="*"',
+                                                         'sudo subscription-manager repos --enable="rhel-7-server-rh-common-rpms" --enable="rhel-7-server-rpms" --enable="rhel-7-server-extras-rpms" --enable="rhel-7-server-ose-3.2-rpms"'
+                                                ])
+      end
+      if nodes_subscription_needed?
+        perform_scp(deployment_master, user, "rhel_subscribe_inventory.yaml", "rhel_subscribe_inventory.yaml")
+      end
+    end
+
+    def subscribe_cluster(deployment_master, user)
+      perform_agent_commands(deployment_master, user, ["sudo mv ~/rhel_subscribe_inventory.yaml /usr/share/ansible/openshift-ansible/"])
+      run_playbook_command(deployment_master, user, "ansible-playbook /usr/share/ansible/openshift-ansible/playbooks/byo/rhel_subscribe.yml -i "\
+                         "/usr/share/ansible/openshift-ansible/rhel_subscribe_inventory.yaml 1> /tmp/ansible.log 2> /tmp/openshift-ansible.log.2 < /dev/null")
+    end
+
+    def nodes_subscription_needed?
+      container_deployment_nodes.count > 1
+    end
+
+    def ssh_user
+      ssh_auth.userid
```

Comment:
@alongoldboim there's no `alias` (or similar) syntactic sugar that you can use here?